packages/plugin-avatar/src/avatar-url-batcher.js

```diff
@@ -0,0 +1,115 @@
+/**!
+ *
+ * Copyright (c) 2015-2016 Cisco Systems, Inc. See LICENSE file.
+ * @private
+ */
+
+import {Batcher} from '@ciscospark/spark-core';
+
+/**
+ * @class AvatarUrlBatcher
+ */
+const AvatarUrlBatcher = Batcher.extend({
+  namespace: `Avatar`,
+
+  /**
+   * Munge the response into items that can be uniquely
+   * identified by uuid + size and process success or failures
+   *
+   * @param {HttpResponseObject} res
+   * @returns {Promise}
+   */
+  handleHttpSuccess(res) {
+    try {
+      return Promise.all(
+        res.options.body.map((req) => Promise.all(
+            req.sizes.map((size) => {
+              const response = res.body[req.uuid] && res.body[req.uuid][size] || undefined;
+              return this.acceptItem(Object.assign({}, {uuid: req.uuid, size, response}));
+            })
```

Comment:
response.options.body contains the initial request body, an array of uuids that each contain an array of requested sizes.

Response.body contains a map of uuids to an array of maps of <size, url> tuples
This code iterates over the request and pulls the corresponding <uuid, size> response, passing each througfh acceptItem to complete the deferred promise